client/js/Core/componentsandbox.js

```diff
@@ -261,16 +263,261 @@ SCWeb.core.ComponentSandbox.prototype.createViewersForScStructs = function (cont
+                            new sc.ScAddr(sourceElem),
+                        );
+                        
+                        if (!visitedElements.includes(targetElem) && !levelNodes.includes(targetElem) || !visitedElements.includes(sourceElem) && !levelNodes.includes(sourceElem)) {
+                                
+                            levelNodes.push(targetElem);
+                            visitedElements.push(targetElem);
+                            levelNodes.push(sourceElem);
+                            visitedElements.push(sourceElem);
+
+                            self.eventStructUpdate(true, triple.get("src").value, triple.get("edgeFromContourToSecondElem").value, scale);
+                        };
+
+                        let resultEdgeElementsEdges = await window.scClient.templateSearch(scTemplateSearchTargetNodes);
+
+                        for (triple of resultEdgeElementsEdges) {
+                            self.eventStructUpdate(true, triple.get("src").value, triple.get("edgeFromContourToSourceElem").value, scale);
+                            self.eventStructUpdate(true, triple.get("src").value, triple.get("edgeFromContourToTargetElem").value, scale);
+                        };
+                    };
+                    edgeToEdge = false;
+                    continue;
+                };
+
+                if (!visitedElements.includes(secondElem) && !levelNodes.includes(secondElem)) {
+                    levelNodes.push(secondElem);
+                    visitedElements.push(secondElem);
+                    self.eventStructUpdate(true, triple.get("src").value, triple.get("edgeFromContourToSecondElem").value, scale);
+                    self.eventStructUpdate(true, triple.get("src").value, triple.get("edgeFromContourToMainEdge").value, scale);
+                };
+                
+                if (!levelNodes.length) continue;
+                if (secondElem === mainElements[0]) continue;
+
+                if (withRelation) {
+                    if (scAddr === mainElements[0] && edgeToEdge) continue;
+                    let relationNode = triple.get("relationNode").value;
+
+                    if (relationNodes.includes(relationNode)) continue;
+
+                    if (!visitedElements.includes(relationNode) && !levelNodes.includes(relationNode)) {
+                        levelNodes.push(relationNode);
+                        visitedElements.push(relationNode);
+                    }
+                    self.eventStructUpdate(true, triple.get("src").value, triple.get("edgeFromContourToRelationNode").value, scale);
+                    self.eventStructUpdate(true, triple.get("src").value, triple.get("edgeFromContourToEdgeFromRelationNodeToedgeFromMainNodeToSecondElem").value, scale);
+                };
+            };
+            return [...new Set(levelNodes)];
+        };
+
+        searchAllLevelEdges([mainElements], levelScales, 1, [...mainElements]);
```

Comment:
why 1? whats that mean?

Reply:
find this function and see the arguments